### ISSUE TITLE
[MAUI] Padding on UnderlinedTabItem leads to inexplicable behavior

### ISSUE THREAD

**Issue by dima059** [roubachof/Sharpnado.Tabs#111]
**Platform:**
 - OS: [Android]
 - Device: [Samsung S22 Ultra (Android 14) and Pixel 6 (Android 15 Preview)]
 - Sdk vervion: [Android SDK 34]
 - MAUI: [8.0.300]

**Describe the bug**
When adding Padding to UnderlinedTabItem in TabHostView (TabType="Scrollable"), tabs starting to move and grow. And this does not happen on all devices.

**To Reproduce**
1. I have used MauiSample to reproduce this: https://github.com/roubachof/Sharpnado.Tabs/blob/main/MauiSample/
2. Add `Padding="5"` to any `tabs:UnderlinedTabItem` tag (on Views/TabM.xaml)
3. See error

**Video**

https://github.com/roubachof/Sharpnado.Tabs/assets/36166982/f4606cb9-b847-4cda-9d3f-8dcc00f29573

**Comment by dima059** [comment]
I found a workaround: adding Margin="1, 0" fixes this behavior. With 1px margin you can get padding work as expected.

But i still think it is a problem.

**Comment by roubachof** [comment]
Yeah I repro that, I guess it's a maui bug with grid and the Auto size...

**Comment by roubachof** [comment]
https://github.com/dotnet/maui/issues/25499

**Comment by roubachof** [comment]
You can even put Margin="0.5,0" :)
